app/helpers/panel_helper.rb

```diff
@@ -0,0 +1,25 @@
+module PanelHelper
+  def render_panels(user = panel_user, &block)
+    options = { vertical_split: !user.full_width_code_panes? }
+
+    content_tag :div, capture(&block), class: panel_classes(options)
+  end
+
+  private
+
+  def panel_classes(vertical_split:)
+    classes = ["panels"]
+
+    classes << "panels--vertical-split" if vertical_split
+
+    classes.join(" ")
+  end
+
+  def panel_user
+    current_user || guest_user
+  end
+
+  def guest_user
```

Comment:
I like this approach think this is quite risky, because all helper methods get merged into views, so if we have `guest_user` in another helper class, this method will get overriden unexpectedly.

I think this would be simpler as this:

```ruby
  def render_panels(user = nil, &block)
    options = { vertical_split: !user.try(&:full_width_code_panes?) }

    content_tag :div, capture(&block), class: panel_classes(options)
  end
```

which would mean both of the helper methods could be removed.

Reply:
Ah. Yes. Good point in bringing up that this will be mixed into the views.